### ISSUE TITLE
Ability to set a default profile when opening links--And/or prompt for a profile

### ISSUE THREAD

**Issue by techlore-official** [brave/brave-browser#17251]
## Description 
User profiles are a great way to compartmentalize things within the same browser. However, there is no way for a user to consciously choose which profile to use when opening a link from an external program. This can be a pretty frustrating issue for users who need to maintain OPSEC between profiles. 

### Example 

A friend messages you on Signal a TikTok link, something you want opened in a hardened profile. You click it, but Brave opens the invasive link inside your personal profile. 

### Solutions
Two easy solutions. They can both be integrated and are similar to how mobile operating systems handle default linking to applications. 
1. Simply allow a user to select a default profile. This would allow users to select their more "hardcore" profile as the default to mitigate this issue. 
2. Allow the user to be prompted each time a link is clicked so they always have an option between their profiles before the URL is loaded.

**Comment by bowlbear** [comment]
+1, this would save seconds upon seconds of time and stupid mistakes.

**Comment by TomDotCom4** [comment]
I would be very happy to see this on the Brave browser. Because of the increase in friction for opening a link in a specific profile vs simply clicking it and opening it wherever it goes, I become discouraged by compartmentalising my browsing in the first place which compromises my privacy online.

A feature like this would be quite a game changer for me if implemented. Perhaps even in addition to the above, there could be a setting where certain websites could be made so that they automatically open in one profile while other websites open elsewhere.

**Comment by kjozwiak** [comment]
Quickly checked on the latest Chrome using `92.0.4515.131 (Official Build) (64-bit) (cohort: 92_win_131)` and could also reproduce the above issue. Definitely a chromium issue and not specific to Brave.

BTW, +1 from me as well as I use multiple profiles and run into the above pretty often.

**Comment by jasonbcox** [comment]
Relevant Chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=813612

**Comment by heinthant-at-dll** [comment]
Yeah ... but chromium issue seemed inactive. Last one is Nov 19. I want that feature too, since I have personal profile, work profile. Sometimes, when I clicked a link from terminal or something, Work profile is launched instead of personal one.

**Comment by tpst** [comment]
Any updates on this? This would be an extremely useful feature

**Comment by MariojGG** [comment]
+1

**Comment by IkelAtomig** [comment]
It would be better to do like Microsoft edge's Automatic profile swithing.

**Comment by MariojGG** [comment]
Any news about this?

**Comment by jasonpatrickellykrause** [comment]
> It would be better to do like Microsoft edge's Automatic profile swithing.

Agree, Edge has this feature down.
![image](https://user-images.githubusercontent.com/8408603/224297965-74a79f9e-4233-4b95-b17a-f381f75657d2.png)

**Comment by radiohe4d** [comment]
I was able to make this work on Windows 11.

**Steps:**
- Get the directory name of the profile you want to have open links by default.
  - Open your default/main profile and type ```brave://version``` into the address bar.
  - Look at the "Profile Path" property. The last part of that URL should be your profile directory. (e.g "Profile 1")

- Update your registry:
  - Go to ```Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\BraveHTML\shell\open\command```
  - Add ```-profile-directory="Profile 1"``` to the end of the existing value in the registry, changing "Profile 1" to the directory name you found in the previous steps.

- Update the Brave shortcut that windows uses:
  - Open the windows search and search for "brave".
  - Right click it and select "Open file location".
  - Right click the brave shortcut and select "Properties"
  - In the "Target" field, add ```-profile-directory="Profile 1"``` to the end of the field, again, make sure to replace with your profile directory name.
  - Hit "Apply" and "Ok".

**Possible Pitfalls:**
- Make sure the correct characters are being copied and pasted. Sometimes the dash ```-``` and the quotes ```""``` that are copied from websites are not the correct character set. If you are having trouble, try retyping those characters manually.

- The Brave/Chromium parameters have two formats. I believe the command line/console params use two dashes ```--profile-directory``` where as the file path & registry params use a single dash ```-profile-directory```. Again double check you are using the correct version.

**Conclusion:**

Now anytime I open a link outside of the browser, it opens in my main brave profile, even if I have opened a different profile before hand. I haven't done extensive testing, so it may not work in all scenarios.

Hopefully this helps the rest of you desperate for this feature like I was. It would be great to see it as an actual feature in Brave as the security concerns around mixing cookies is very real.

**Comment by alienator88** [comment]
Mac Workaround:

1. Install [Hammerspoon](https://www.hammerspoon.org/) or if already using it continue 
2. Set default web browser in you mac Settings/Preferences to `Hammerspoon`
3. Download the [URLDispatcher](https://github.com/Hammerspoon/Spoons/raw/master/Spoons/URLDispatcher.spoon.zip) spoon, unzip it and double click the spoon file to install it in Hammerspoon
4. Add the below code to your `init.lua` script that Hammerspoon loads on launch
```
hs.loadSpoon("URLDispatcher")

local sendToProfileBrave = function(t)
    local fn = function(url)
        local t = hs.task.new("/Applications/Brave Browser.app/Contents/MacOS/Brave Browser", nil, function()
            return false
        end, {"--profile-directory=" .. t[2], url})
        t:start()
    end
    return {t[1], nil, fn}
end

spoon.URLDispatcher.url_patterns = {sendToProfileBrave {"", "Profile 1"}}
spoon.URLDispatcher:start()
```
5. Run Hammerspoon

This will set Hammerspoon as your default external http/https link handler. 
You can set the profile name you want to launch links in from line: `spoon.URLDispatcher.url_patterns = {sendToProfileBrave {"", "Profile 1"}}`
In my case, I'm launching **Profile 1**.

**Comment by opoto** [comment]
A workaround for Windows: https://github.com/nref/BrowseRouter

**Comment by Antglo** [comment]
> +1, this would save seconds upon seconds of time and stupid mistakes.

It might save seconds. Not sure if that is sarcasm, but try joining any kind of Google Meet. Even if the G-meet link is sent to the correct email. Brave still tries to enforce my other profile upon clicking the meet link, meaning I have to join the meet as a "guest". I am hoping there is a solution to this soon!

P.S. I have to join as a "guest" even after switching back to the profile that the link originated from.

**Comment by TheRealPSV** [comment]
Are there any plans to fix this? It's actually a big annoyance, considering if I click a link I want it to open in my main profile where I'm not logged into anything, rather than in my secondary profile where I'm logged into, for example, Google, and now whatever I clicked on is being tracked by Google.

**Comment by kengribble** [comment]
For some reason I can't even get Brave to use the last profile I used, it keeps picking the profile I don't want any external URL to open in.

This feature seems like a "must have" for Profile use. What will it take to get it moved forward?

**Comment by SNgari** [comment]
An additional requirement to this request is that what's opened on the other other profile is always a clean link. 

Currently we have the option to open link as profile 2, but I'm not sure whether the clean link is used or just the 'dirty' one

**Comment by notarib-catcher** [comment]
+1 Accidentally logged in on steam on my work profile, because I happened to commit the sin of accidentally checking that mid steam sale scrolling session. need this to prevent stupid mistakes
